.github/workflows/create-agentic-github-prerelease.yml

```diff
@@ -0,0 +1,164 @@
+                      REMAINDER=$(echo "$BRANCH_NAME" | sed 's/^feature\///')
+                      echo "TAG_NAME=agentic-pre-$REMAINDER" >> $GITHUB_ENV
+                      echo "PRERELEASE_NAME=$REMAINDER" >> $GITHUB_ENV
+                  elif [[ "$BRANCH_NAME" == release/agentic/* ]]; then
+                      REMAINDER=$(echo "$BRANCH_NAME" | sed 's/^release\/agentic\///')
+                      echo "TAG_NAME=agentic-rc-$REMAINDER" >> $GITHUB_ENV
+                      echo "PRERELEASE_NAME=rc" >> $GITHUB_ENV
+                  else
+                      echo "Error: creating agentic releases for this branch is not supported"
+                      exit 1
+                  fi
+
+            # Make a sever version that is "decorated" as prerelease
+            - name: Create SERVER_VERSION
+              run: |
+                  # example: 1.0.999-pre-main.commitid
+                  # SERVER_VERSION - we're making "imitation" manifests that are accessible
+                  # from GitHub releases, as a convenience for plugins to easily consume
+                  # test/development builds. The version is pulled from the agenticChat field
+                  # in the version.json file.
+
+                  AGENTIC_VERSION=$(jq -r '.agenticChat' app/aws-lsp-codewhisperer-runtimes/src/version.json)
+                  COMMIT_SHORT=$(echo "${{ github.event.workflow_run.head_sha }}" | cut -c1-8)
+                  echo "SERVER_VERSION=$AGENTIC_VERSION-$PRERELEASE_NAME.$COMMIT_SHORT" >> $GITHUB_ENV
+
+            - name: Export outputs
+              id: build
+              run: |
+                  # tag name is the git tag that the github release is linked with
+                  echo "tagname=$TAG_NAME" >> $GITHUB_OUTPUT
+                  #  pre-release name is the semver pre-release decorator (eg 'alpha', 'rc', ...)
+                  echo "prereleasename=$PRERELEASE_NAME" >> $GITHUB_OUTPUT
+                  echo "serverversion=$SERVER_VERSION" >> $GITHUB_OUTPUT
+
+    create-release:
+        runs-on: ubuntu-latest
+        if: ${{ github.event.workflow_run.conclusion == 'success' }}
+        needs: [setup-vars]
+
+        env:
+            #
+            # For `gh` cli.
+            #
+            GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+            TAG_NAME: ${{ needs.setup-vars.outputs.tagname }}
+            #
+            # Used in release_notes.md and git tag
+            #
+            BRANCH: ${{ github.event.workflow_run.head_branch }}
+            COMMIT_ID: ${{ github.event.workflow_run.head_sha }}
+        permissions:
+            contents: write
+
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v4
+              with:
+                  ref: ${{ github.event.workflow_run.head_sha }}
+
+            - name: Setup Node.js
+              uses: actions/setup-node@v4
+              with:
+                  node-version: '20'
```

Comment:
we are delivering node 18 to customers, should we use node 18 instead?

Reply:
Internally we're using node 20 to produce the artifacts, so I kept this the same. I have node 18 installed on my system, and I noticed that the webpack bundling produces warnings because it is expecting node 20, so I figured I'd try to keep things consistent with the existing build system.